agent/src/main/java/com/teamscale/jacoco/agent/testimpact/TestwiseCoverageAgent.java

```diff
@@ -234,9 +237,15 @@ private String error(Response response, String message, Object... arguments) {
 		return message;
 	}
 
-	/** Logs errors that occurred during agent initialization. */
-	private void logInitException(Exception e) {
+	/**
+	 * Logs errors that occurred during agent initialization. Since Spark starts
+	 * the server asynchronously, we just get a callback to this method, and the
+	 * only thing we can do to not lose data unknowingly is terminating the
+	 * process.
+	 */
+	private void logInitExceptionAndExit(Exception e) {
 		logger.error("Agent initialization failed", e);
+		System.exit(1);
```

Comment:
Not sure if this is the best way to handle this. I would have expected the agent to continue with the portscan from where it left if the initialization failed. I fear that this behaviour could lead to very hard to debug scenarios, but maybe I'm wrong. I'm not very familiar with micro-service setups ;-)